File: mvvm-java/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchActivity.java
package com.wasisto.githubuserfinder.android.ui.search;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModel;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.wasisto.githubuserfinder.android.DefaultServiceLocator;
import com.wasisto.githubuserfinder.android.R;
import com.wasisto.githubuserfinder.android.databinding.ActivitySearchBinding;
import com.wasisto.githubuserfinder.android.ui.common.ViewModelFactory;
import com.wasisto.githubuserfinder.android.ui.userdetails.UserDetailsActivity;

public class SearchActivity extends AppCompatActivity {

    private SearchViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ViewModelFactory viewModelFactory = new ViewModelFactory(DefaultServiceLocator.getInstance(getApplication()));
        viewModel = viewModelFactory.create(SearchViewModel.class);

        ActivitySearchBinding binding = ActivitySearchBinding.inflate(getLayoutInflater());
        binding.setViewModel(viewModel);
        binding.setLifecycleOwner(this);
        setContentView(binding.getRoot());

        viewModel.getOpenUserDetailsActivityEvent().observe(this, event -> {
            String username = event.getContentIfNotHandled();
            if (username != null) {
                startActivity(
                        new Intent(this, UserDetailsActivity.class) {{
                            putExtra(UserDetailsActivity.EXTRA_USERNAME, username);
                        }}
                );
            }
        });

        viewModel.getShowErrorToastEvent().observe(this, event -> {
            if (!event.hasBeenHandled()) {
                Toast.makeText(this, R.string.an_error_occurred, Toast.LENGTH_SHORT).show();
            }
        });

        viewModel.load();
    }
}

File: mvvm-kotlin/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchActivity.kt
package com.wasisto.githubuserfinder.android.ui.search

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.activity.viewModels
import com.wasisto.githubuserfinder.android.DefaultServiceLocator
import com.wasisto.githubuserfinder.android.R
import com.wasisto.githubuserfinder.android.databinding.ActivitySearchBinding
import com.wasisto.githubuserfinder.android.ui.common.ViewModelFactory
import com.wasisto.githubuserfinder.android.ui.userdetails.UserDetailsActivity

class SearchActivity : AppCompatActivity() {

    private val viewModel: SearchViewModel by viewModels {
        ViewModelFactory(DefaultServiceLocator.getInstance(application))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(
            ActivitySearchBinding.inflate(layoutInflater).apply {
                viewModel = this@SearchActivity.viewModel
                lifecycleOwner = this@SearchActivity
            }.root
        )

        viewModel.openUserDetailsActivityEvent.observe(this, { event ->
            event.getContentIfNotHandled()?.let { username ->
                startActivity(
                    Intent(this, UserDetailsActivity::class.java).apply {
                        putExtra(UserDetailsActivity.EXTRA_USERNAME, username)
                    }
                )
            }
        })

        viewModel.showErrorToastEvent.observe(this, { event ->
            if (!event.hasBeenHandled) {
                Toast.makeText(this@SearchActivity, R.string.an_error_occurred, Toast.LENGTH_SHORT).show()
            }
        })

        viewModel.load()
    }
}